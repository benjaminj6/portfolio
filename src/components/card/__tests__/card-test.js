/* eslint-env jest */
import { h } from 'preact'
import { shallow } from 'preact-render-spy'

import Card from '../'

test('Card with contents snap', () => {
  const tree = shallow(<Card>Test content</Card>)
  expect(tree).toMatchSnapshot()
})

test('Card with header snap', () => {
  const tree = shallow(<Card header={() => <h1>header</h1>}>test content</Card>)
  expect(tree).toMatchSnapshot()
})
