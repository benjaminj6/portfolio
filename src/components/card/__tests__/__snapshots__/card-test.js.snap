// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card with contents snap 1`] = `
preact-render-spy (1 nodes)
-------
<article class="jsx-1991709517 jsx-1991709517 Card">
  <main class="jsx-1991709517 jsx-1991709517">Test content</main>
  <JSXStyle
    styleId="1991709517"
    css=".jsx-1991709517{--color:var(--gray-normal);}.Card.jsx-1991709517{border-radius:2px;border:1px solid var(--color);background:var(--white);}header.jsx-1991709517{border-bottom:1px solid var(--color);}header.jsx-1991709517,main.jsx-1991709517{padding:1.25rem 1rem;}"
  >
  </JSXStyle>
</article>

`;

exports[`Card with header snap 1`] = `
preact-render-spy (1 nodes)
-------
<article class="jsx-1991709517 jsx-1991709517 Card">
  <header class="jsx-1991709517 jsx-1991709517">function header() {return _ref2;}</header>
  <main class="jsx-1991709517 jsx-1991709517">test content</main>
  <JSXStyle
    styleId="1991709517"
    css=".jsx-1991709517{--color:var(--gray-normal);}.Card.jsx-1991709517{border-radius:2px;border:1px solid var(--color);background:var(--white);}header.jsx-1991709517{border-bottom:1px solid var(--color);}header.jsx-1991709517,main.jsx-1991709517{padding:1.25rem 1rem;}"
  >
  </JSXStyle>
</article>

`;
